File: src/main/java/ru/xpendence/topjavagraduation/entity/Restaurant.java
package ru.xpendence.topjavagraduation.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.SQLRestriction;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "restaurants")
@Getter
@Setter
public class Restaurant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @OneToMany(mappedBy = "restaurant", fetch = FetchType.EAGER)
    private Set<Dish> dishes = new HashSet<>();

    @SQLRestriction("date = current_date()")
    @OneToMany(mappedBy = "restaurant", fetch = FetchType.EAGER)
    private Set<Vote> votes = new HashSet<>();

    public static void enrichForUpdate(Restaurant forUpdate, Restaurant stored) {
        stored.name = forUpdate.name;
    }
}

File: src/main/kotlin/ru/xpendence/topjavagraduation/entity/Restaurant.kt
package ru.xpendence.topjavagraduation.entity

/**
 * Ресторан.
 */
data class Restaurant(
    val id: Long? = null,
    val name: String? = null,
) {
    val dishes: MutableSet<Dish> = hashSetOf()
    val votes: MutableSet<Vote> = hashSetOf()

    companion object {
        @JvmStatic
        fun enrichForUpdate(forUpdate: Restaurant, stored: Restaurant): Restaurant =
            stored.copy(name = forUpdate.name).also { copy ->
                copy.dishes.addAll(stored.dishes)
                copy.votes.addAll(stored.votes)
            }
    }
}
